plugin/src/test/java/org/opensearch/ml/rest/RestSearchIndexToolIT.java

```diff
@@ -105,7 +106,13 @@ public void testSearchIndexToolInFlowAgent_withEmptyIndexField_thenThrowExceptio
             + "  }\n"
             + "}\n";
         Exception exception = assertThrows(ResponseException.class, () -> executeAgent(agentId, agentInput));
-        MatcherAssert.assertThat(exception.getMessage(), containsString("SearchIndexTool's two parameter: index and query are required!"));
+        String exceptionMessage = exception.getMessage();
+        int numNodes = NumberUtils.createInteger(System.getProperty("cluster.number_of_nodes", "1"));
+        if (numNodes == 1 || !exceptionMessage.contains("RemoteTransportException")) {
+            // in multi-node cluster, the exception maybe of type RemoteTransportException
+            // we cannot get the concrete exception message from RemoteTransportException
```

Comment:
Can you try `getDetailedMessage` method  of `RemoteTransportException`  ? I think it could get detailed message

Reply:
Unfortunately, `getDetailedMessage` method is only available for `OpenSearchException`. Here, the exception is always of type `ResponseException` (even in the multi cluster test).

Reply:
I will update the comment to avoid confusing people.